tensorflow/tools/ci_build/linux/rocm/rocm_ci_sanity.sh

```diff
@@ -0,0 +1,547 @@
+#!/usr/bin/env bash
```

Comment:
What is the rationale for having a ROCm version of the `ci_sanity.sh` (as opposed to re-using the existing one)? We will need the answer for this question if/when we try to upstream this change.

Reply:
one reason is hinted to, in the PR description. The `ci_sanity.sh` script runs a few more checks than the `rocm_ci_sanity.sh` script. The way the script is structured it will run all the checks contained within it, and we cannot pick and choose the checks we want to run via command line arguments, and hence the need for a separate script for ROCm

Another reason is some ROCm specific tweaks (for e.g. the new `get_changed_files_for_this_branch` routine)